pennylane/ops/op_math/symbolicop.py

```diff
@@ -136,3 +138,102 @@ def map_wires(self, wire_map: dict):
         new_op = copy(self)
         new_op.hyperparameters["base"] = self.base.map_wires(wire_map=wire_map)
         return new_op
+
+
+class ScalarSymbolicOp(SymbolicOp):
+    """Developer-facing base class for single-operator symbolic operators that contain a
+    scalar coefficient.
+
+    Args:
+        base (~.operation.Operator): the base operation that is modified symbolicly
+        coeff (float): the scalar coefficient
+        do_queue (bool): indicates whether the operator should be recorded when created in a tape
+            context
+        id (str): custom label given to an operator instance, can be useful for some applications
+            where the instance has to be identified
+
+    This *developer-facing* class can serve as a parent to single base symbolic operators, such as
+    :class:`~.ops.op_math.SProd` and :class:`~.ops.op_math.Pow`.
+    """
+
+    _name = "CoeffSymbolicOp"
+
+    def __init__(self, base, scalar: float, do_queue=True, id=None):
+        self.scalar = scalar
+        super().__init__(base, do_queue=do_queue, id=id)
+        self._batch_size = self._check_and_compute_batch_size(scalar)
+
+    @property
+    def batch_size(self):
+        return self._batch_size
+
+    def _check_and_compute_batch_size(self, scalar):
+        scalar_size = qml.math.size(scalar)
+        if scalar_size == 1:
+            # coeff is not batched
```

Comment:
Like mentioned above, this might break the convention on whether we call something with shape `(1,)` that is expected to be a scalar broadcasted or not.

Reply:
Changed!